pkg/download/retry.go

```diff
@@ -24,21 +25,36 @@ const (
 	expRetryN = 7 // how many times we retry the Download
 	expRetryK = time.Second * 3
 	expRetryM = 2
+	writeBufSize = 1024 * 8
 )
 
 // WithRetries retrieves a response body using the specified downloader. Any
 // error returned from d will be retried (and retrieved response bodies will be
 // closed on failures). If the retries do not succeed, the last error is returned.
 //
 // It sleeps in exponentially increasing durations between retries.
-func WithRetries(ctx *log.Context, downloaders []Downloader, sf SleepFunc) (io.ReadCloser, error) {
+func WithRetries(ctx *log.Context, f *File, downloaders []Downloader, sf SleepFunc) (int64, error) {
```

Comment:
Please double check whether there are other callers for this package. I believe RunCommand V2 may use it, in which case it will break when the pkg is next imported.

Reply:
Imports are versioned, importers can get away with not having to change their code as long as don't update their reference to this repo in their go.mod file via some command like "go get -u github.com/Azure/custom-script-extension-linux"

Reply:
Looks like other extensions have their own implementations of Save, Download, and WithRetries methods  -

https://github.com/Azure/run-command-extension-linux/blob/master/pkg/download/save.go
https://github.com/Azure/run-command-extension-linux/blob/master/pkg/download/retry.go

https://github.com/Azure/run-command-handler-linux/blob/main/pkg/download/save.go
https://github.com/Azure/run-command-handler-linux/blob/main/pkg/download/retry.go